redfish-core/lib/log_services.hpp

```diff
@@ -6382,11 +6382,44 @@ inline void getRedfishUriByDbusObjPath(
         // Fill the all parents Redfish URI id.
         // For example, the processors id for the core.
         // "/redfish/v1/Systems/system/Processors/<str>/SubProcessors/core0"
+        std::vector<std::pair<RedfishResourceDBusInterfaces, size_t>>
+            ancestorsIfaces;
+        while (uriIdPos != std::string::npos)
+        {
+            std::string parentRedfishUri = redfishUri.substr(0, uriIdPos - 1);
+            RedfishUriListType::const_iterator parentRedfishUriIt =
+                std::find_if(redfishUriList.begin(), redfishUriList.end(),
+                             [parentRedfishUri](const auto& ele) {
+                return parentRedfishUri == ele.second;
+            });
+
+            if (parentRedfishUriIt == redfishUriList.end())
+            {
+                BMCWEB_LOG_ERROR(
+                    "Failed to fill Links:OriginOfCondition because unable to get parent Redfish URI [{}] DBus interface for the identified Redfish URI: {} of the given DBus object path: {}",
+                    parentRedfishUri, redfishUri, dbusObjPath.str);
+                messages::internalError(asyncResp->res);
+                return;
+            }
+            ancestorsIfaces.emplace_back(parentRedfishUriIt->first, uriIdPos);
+            size_t tempUriIdPos = uriIdPos - uriIdPattern.length();
+            uriIdPos = redfishUri.rfind(uriIdPattern, tempUriIdPos);
+        }
+
+        // GetAncestors only accepts "as" for the interface list
+        std::vector<RedfishResourceDBusInterfaces> ancestorsIfacesOnly;
+        std::transform(
+            ancestorsIfaces.begin(), ancestorsIfaces.end(),
+            std::back_inserter(ancestorsIfacesOnly),
+            [](const std::pair<RedfishResourceDBusInterfaces, size_t>& p) {
+            return p.first;
+        });
```

Comment:
Similar `std::find_if` case, `std::ranges` is preferred to use.

```
        std::transform(
            ancestorsIfaces, std::back_inserter(ancestorsIfacesOnly),
            [](const std::pair<RedfishResourceDBusInterfaces, size_t>& p) {
            return p.first;
        });
```

Reply:
Sure. Will address this after the cherry-picking is completed, as there are few more commits letft from, https://github.com/ibm-openbmc/bmcweb/pull/542/commits and they are looking at getting everything into 1110 at the earliest

Reply:
Would you add a JIRA story and add the link here?